.github/actions/go-setup/action.yml

```diff
@@ -151,17 +151,26 @@ runs:
         echo "Modules size:"
         du -hd0 "$GODIRS/modcache"
 
+    - name: Install git-restore-mtime
+      shell: bash
+      run: |
+        path="$(go env GOBIN)/git-restore-mtime"
+        curl -o "$path" https://raw.githubusercontent.com/MestreLion/git-tools/v2022.12/git-restore-mtime
+        chmod +x "$path"
```

Comment:
this solution has several imperfections that did not bother me:
- It has nothing to do with GOBIN, but we need some PATH that is the same on all OS.
- It could be downloaded only when it does not exist, as you write, but it is very fast. If you want, improve it.